src/devices/cpu/arm7/arm7.h

```diff
@@ -614,6 +614,7 @@ class arm946es_cpu_device : public arm9_cpu_device
 public:
 	// construction/destruction
 	arm946es_cpu_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock);
+	arm946es_cpu_device(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, uint32_t clock);
```

Comment:
Trampoline constructors should be protected, not public - if something other than a derived class can set the device type, it breaks basic assumptions in the framework.